drain-service/drain_training_inferencing.py

```diff
@@ -294,44 +293,57 @@ def weighted_avg_and_std(values, weights):
 
             if (
                 weighted_vol < 0.199
-                and training_start_ts_ns != very_first_ts_ns
+                and training_start_ts_ms != very_first_ts_ns
                 and train_on_next_chance
             ):
-                training_start_ts_ns = time.time_ns()
+                training_start_ts_ms = int(
+                    pd.to_datetime("now", utc=True).timestamp() * 1000
+                )
 
             if weighted_vol > 0.155 and not train_on_next_chance and stable:
-                training_end_ts_ns = time.time_ns()
+                training_end_ts_ms = int(
+                    pd.to_datetime("now", utc=True).timestamp() * 1000
+                )
                 normal_periods.append(
-                    {"start_ts": training_start_ts_ns, "end_ts": training_end_ts_ns}
+                    {"start_ts": training_start_ts_ms, "end_ts": training_end_ts_ms}
                 )
                 stable = False
-                training_start_ts_ns = -1.0
+                training_start_ts_ms = -1.0
 
             if weighted_vol <= 0.15 and (
                 train_on_next_chance
                 or num_drain_templates > (2 * num_templates_in_last_train)
             ):
                 num_templates_in_last_train = num_drain_templates
                 logging.info(f"SENDING TRAIN SIGNAL on iteration {iteration}")
-                if training_start_ts_ns != -1.0:
-                    training_end_ts_ns = time.time_ns()
+                if training_start_ts_ms != -1.0:
+                    training_end_ts_ms = int(
+                        pd.to_datetime("now", utc=True).timestamp() * 1000
+                    )
                     normal_periods.append(
-                        {"start_ts": training_start_ts_ns, "end_ts": training_end_ts_ns}
+                        {"start_ts": training_start_ts_ms, "end_ts": training_end_ts_ms}
                     )
                 train_payload = {
-                    "model_to_train": "nulog",
-                    "time_intervals": normal_periods,
+                    "start_ts": training_start_ts_ms,
+                    "end_ts": training_end_ts_ms,
                 }
+                try:
+                    await es.index(index="opni-normal-intervals", body=train_payload)
```

Comment:
Where are you creating the opni-normal-intervals index? Don't you need to follow this pattern first to create the index before you place payload into the index?
```
es.indices.create(index="opni-normal-intervals", ignore=400)
```

Reply:
It's set up here: https://github.com/rancher/opni-kibana-dashboard/pull/9